HARK/ConsumptionSaving/ConsWealthUtilityModel.py

```diff
@@ -0,0 +1,1286 @@
+    """
+    cFunc_terminal = ConstantFunction(0.0)
+    vFunc_terminal = ConstantFunction(0.0)
+    vPfunc_terminal = ConstantFunction(0.0)
+    vPPfunc_terminal = ConstantFunction(0.0)
+    solution_terminal = ConsumerSolution(
+        cFunc=cFunc_terminal,
+        vFunc=vFunc_terminal,
+        vPfunc=vPfunc_terminal,
+        vPPfunc=vPPfunc_terminal,
+        mNrmMin=ConstantFunction(0.0),
+        hNrm=ConstantFunction(0.0),
+        MPCmin=1.0,
+        MPCmax=1.0,
+    )
+    solution_terminal.hLvl = solution_terminal.hNrm
+    solution_terminal.mLvlMin = solution_terminal.mNrmMin
+    return solution_terminal
+
+
+class ChiFromOmegaFunction:
+    """
+    A class for representing a function that takes in values of omega = EndOfPrdvPnvrs / aNrm
+    and returns the corresponding optimal chi = cNrm / aNrm. The only parameters
+    that matter for this transformation are the coefficient of relative risk
+    aversion (rho) and the share of wealth in the Cobb-Douglas aggregator (delta).
+
+    Parameters
+    ----------
+    CRRA : float
+        Coefficient of relative risk aversion.
+    WealthShare : float
+        Share for wealth in the Cobb-Douglas aggregator in CRRA utility function.
+    N : int, optional
+        Number of interpolating gridpoints to use (default 501).
+    z_bound : float, optional
+        Absolute value on the auxiliary variable z's boundary (default 15).
+        z represents values that are input into a logit transformation
+        scaled by the upper bound of chi, which yields chi values.
+    """
+
+    def __init__(self, CRRA, WealthShare, N=501, z_bound=15):
+        self.CRRA = CRRA
+        self.WealthShare = WealthShare
+        self.N = N
+        self.z_bound = z_bound
+        self.update()
+
+    def f(self, x):
+        """
+        Define the relationship between chi and omega, and evaluate on the vector
+        """
+        r = self.CRRA
+        d = self.WealthShare
+        return x ** (1 - d) * ((1 - d) * x ** (-d) - d * x ** (1 - d)) ** (-1 / r)
+
+    def update(self):
+        """
+        Construct the underlying interpolation of log(omega) on z.
+        """
+        # Make vectors of chi and z
+        chi_limit = (1.0 - self.WealthShare) / self.WealthShare
+        z_vec = np.linspace(-self.z_bound, self.z_bound, self.N)
+        exp_z = np.exp(z_vec)
+        chi_vec = chi_limit * exp_z / (1 + exp_z)
+
+        omega_vec = self.f(chi_vec)
+        log_omega_vec = np.log(omega_vec)
+
+        # Construct the interpolant
+        zFromLogOmegaFunc = LinearInterp(log_omega_vec, z_vec, lower_extrap=True)
+
+        # Store the function and limit as attributes
+        self.func = zFromLogOmegaFunc
+        self.limit = chi_limit
+
+    def __call__(self, omega):
+        """
+        Calculate optimal values of chi = cNrm / aNrm from values of omega.
+
+        Parameters
+        ----------
+        omega : np.array
+            One or more values of omega = EndOfPrdvP / aNrm.
```

Comment:
This `utility()` function is duplicated in `ConsWealthPortfolioModel.py`. Consider extracting these utility helper functions (`utility`, `dudc`) to a shared location to avoid duplication.

Reply:
Maybe, but this also makes it harder to read and understand the code for one model.